ignition_ros2_control/src/ignition_ros2_control_plugin.cpp

```diff
@@ -0,0 +1,400 @@
+
+      // Make sure the joint type is supported, i.e. it has exactly one
+      // actuated axis
+      const auto * jointType = _ecm.Component<ignition::gazebo::components::JointType>(jointEntity);
+      switch (jointType->Data()) {
+        case sdf::JointType::PRISMATIC:
+        case sdf::JointType::REVOLUTE:
+        case sdf::JointType::CONTINUOUS:
+        case sdf::JointType::GEARBOX:
+          {
+            // Supported joint type
+            break;
+          }
+        case sdf::JointType::FIXED:
+          {
+            igndbg << "[ignition_ros2_control] Fixed joint [" << jointName <<
+              "(Entity=" << jointEntity << ")] is skipped.\n";
+            continue;
+          }
+        case sdf::JointType::REVOLUTE2:
+        case sdf::JointType::SCREW:
+        case sdf::JointType::BALL:
+        case sdf::JointType::UNIVERSAL:
+          {
+            ignwarn << "[ignition_ros2_control] Joint [" << jointName <<
+              "(Entity=" << jointEntity <<
+              ")] is of unsupported type. Only joints with a single axis"
+              " are supported.\n";
+            continue;
+          }
+        default:
+          {
+            ignwarn << "[ignition_ros2_control] Joint [" << jointName <<
+              "(Entity=" << jointEntity << ")] is of unknown type.\n";
+            continue;
+          }
+      }
+      output[jointName] = jointEntity;
+    }
+
+    return output;
+  }
+};
+
+//////////////////////////////////////////////////
+IgnitionROS2ControlPlugin::IgnitionROS2ControlPlugin()
+: dataPtr(std::make_unique<IgnitionROS2ControlPluginPrivate>())
+{
+}
+
+//////////////////////////////////////////////////
+void IgnitionROS2ControlPlugin::Configure(
+  const ignition::gazebo::Entity & _entity,
+  const std::shared_ptr<const sdf::Element> & _sdf,
+  ignition::gazebo::EntityComponentManager & _ecm,
+  ignition::gazebo::EventManager &)
+{
+  // Make sure the controller is attached to a valid model
+  const auto model = ignition::gazebo::Model(_entity);
+  if (!model.Valid(_ecm)) {
+    ignerr << "[Ignition ROS 2 Control] Failed to initialize because [" <<
+      model.Name(_ecm) << "(Entity=" << _entity <<
+      ")] is not a model. Please make sure that"
+      " Ignition ROS 2 Control is attached to a valid model.\n";
+    return;
+  }
+
+  // Get params from SDF
+  std::string paramFileName = _sdf->Get<std::string>("parameters");
+
+  if (paramFileName.empty()) {
+    ignerr << "Ignition ros2 controle found an empty parameters file. " <<
+      "Failed to initialize.";
+    return;
+  }
+
+  std::vector<std::string> arguments = {"--ros-args", "--params-file", paramFileName};
+
+  std::vector<const char *> argv;
+  for (const auto & arg : arguments) {
+    argv.push_back(reinterpret_cast<const char *>(arg.data()));
+  }
+
+  if (!rclcpp::ok()) {
+    rclcpp::init(static_cast<int>(argv.size()), argv.data());
+    this->dataPtr->node = rclcpp::Node::make_shared("ignition_ros2_control");
```

Comment:
Do we need to give the node a unique name, in case more than one controller is instantiated at a time?

Reply:
Yes we should do this. We should at least enable to set the name through parameters. This name should be also the prefix for `controller_manager`